added detailed instructions for xtensa cmake and devserver setup

zonglinpeng opened the pull request:
Summary:
Recently change in cadence added new internal dependencies which caused confusion in building the hifi binaries. Added detailed intrustiones in a sample bash script to build with xtensa toolchain.

Also based on internal development, added instructions to setup cmake/pip on internal developement servers.

Reviewed By: hsharma35

Differential Revision: D62768143

mcremon-meta commented on backends/cadence/README.md:
```diff
@@ -4,6 +4,10 @@
 - HiFi Audio
 - ...
 
+## Setup (Internal Only)
```
should we put this in OSS at all?

zonglinpeng replied:
Tentatively yes, because it's instruction for cadence oss but exposes nothing extra

mcremon-meta replied:
maybe I'm reading this wrong, we have an internal link and the arvr and nfs paths there... that's all internal right?

zonglinpeng replied:
Correct. Just rethought I will remove these lines

zonglinpeng committed:
added detailed instructions for xtensa cmake and devserver setup (#5398)

Summary:

Recently change in cadence added new internal dependencies which caused confusion in building the hifi binaries. Added detailed intrustiones in a sample bash script to build with xtensa toolchain.

Also based on internal development, added instructions to setup cmake/pip on internal developement servers.

Reviewed By: hsharma35

Differential Revision: D62768143
